web/hosts.go

```diff
@@ -49,36 +49,37 @@ func NewHostListHandler(hostsService services.HostsService) gin.HandlerFunc {
 			Size:   pageSize,
 		}
 
-		hostList, err := hostsService.GetAll(hostsFilter, page)
+		hostListPaginated, err := hostsService.GetAll(hostsFilter, page)
 		if err != nil {
 			_ = c.Error(err)
 			return
 		}
 
-		filterSIDs, err := hostsService.GetAllSIDs()
+		hostListAll, err := hostsService.GetAll(hostsFilter, nil)
 		if err != nil {
 			_ = c.Error(err)
 			return
 		}
 
-		filterTags, err := hostsService.GetAllTags()
+		filterSIDs, err := hostsService.GetAllSIDs()
 		if err != nil {
 			_ = c.Error(err)
 			return
 		}
 
-		count, err := hostsService.GetCount()
+		filterTags, err := hostsService.GetAllTags()
 		if err != nil {
 			_ = c.Error(err)
 			return
 		}
-		pagination := NewPagination(count, pageNumber, pageSize)
 
-		hContainer := NewHostsHealthContainer(hostList)
+		pagination := NewPagination(len(hostListAll), pageNumber, pageSize)
+
+		hContainer := NewHostsHealthContainer(hostListAll)
```

Comment:
Same as for the cluster comment, we fetch the list twice, while we just need the health count. I understand that queries are to be done to aggregate the data in the service anyway, but fetching the whole list with preloaded data seems overkill. It also helps with clarity in the controller which I feel is doing too much stuff ATM.